overlord/snapstate/snapmgr.go

```diff
@@ -1307,11 +1312,17 @@ func (m *SnapManager) ensureMountsUpdated() error {
 			// TODO refactor so the check for kernel type is not repeated
 			// in the installation case
 			snapType, _ := snapSt.Type()
+			// We cannot ensure for this type yet as the mount unit
+			// flags depend on the model in this case.
+			if snapType == snap.TypeKernel && dev == nil {
+				continue
```

Comment:
Later we set `m.ensuredMountsUpdated = true`. Should we have a special value for which kernel units still need to be updated once we get in the state where we can generate them?

Reply:
Imho that is a bit of an overkill, especially as the case when there is no model is legacy, I think.